src/radical/entk/appman/appmanager.py

```diff
@@ -270,9 +270,28 @@ def run(self):
 
             self._prof.prof('amgr run started', uid=self._uid)
 
+
+            # Submit resource request if not resource allocation done till now or
+            # resubmit a new one if the old one has completed
+            if self._resource_manager:
+                res_alloc_state = self._resource_manager.get_resource_allocation_state()
+                if (not res_alloc_state) or (res_alloc_state in self._resource_manager.get_completed_states()):
+
+                    self._logger.info('Starting resource request submission')
+                    self._prof.prof('init rreq submission', uid=self._uid)
+                    self._resource_manager._submit_resource_request()
+                    res_alloc_state = self._resource_manager.get_resource_allocation_state()
+                    if res_alloc_state in self._resource_manager.get_completed_states():
+                        raise EnTKError(msg="Cannot proceed. Resource allocation ended up in %s"%res_alloc_state)
+
+            else:
+
+                self._logger.exception('Cannot run without resource manager, please create and assign a resource manager')
+                raise EnTKError(text='Missing resource manager')
+
```

Comment:
Why do you have `EntkError(text=...)` and `EntkError(msg=...)`?

Reply:
this was an error, it should be 'msg'. Fixed